### PR TITLE
add email sending

### PR DESCRIPTION
**Pull request by kolesnykv**
To successfully lunch app you need to provide decrypt function in configuration settings of main application 
image
to preview decrypted data in console you need to use following mvn command mvn jasypt:decrypt -Djasypt.encryptor.password="Password that was chosen to encrypt data"-Djasypt.plugin.path="file:/Users/apple/IdeaProjects/knubisoft-pirozochki/application/src/main/resources/application.properties"
For encryption data change decrypt to encrypt
It is also possible to encrypt or decrypt specific value after executing following mvn command mvn jasypt:encrypt-value -Djasypt.encryptor.password=password -Djasypt.plugin.value=Password

### CODE REVIEW

File: application/src/main/java/com/knubisoft/application/service/EmailServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+            mailMessage.setFrom(sender);
+            mailMessage.setTo(emailDetails.getRecipient());
+            mailMessage.setText(emailDetails.getMsgBody());
+            mailMessage.setSubject(emailDetails.getSubject());
+
+            javaMailSender.send(mailMessage);
+            return "Mail Sent Successfully...";
+        } catch (Exception e) {
+            return "Error while Sending Mail";
```
Comment (nikitakoliadin) [inline application/src/main/java/com/knubisoft/application/service/EmailServiceImpl.java]: Pls log an error.

File: application/src/main/java/com/knubisoft/application/service/EmailServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+            FileSystemResource file = new FileSystemResource(
+                    new File(emailDetails.getAttachment()));
+
+            mimeMessageHelper.addAttachment(file.getFilename(), file);
+
+            javaMailSender.send(mimeMessage);
+            return "Mail sent Successfully";
+        } catch (MessagingException e) {
+            return "Error while sending mail!!!";
```
Comment (nikitakoliadin) [inline application/src/main/java/com/knubisoft/application/service/EmailServiceImpl.java]: The same

File: application/src/main/resources/application.properties [hunk 3]
```diff
@@ -42,4 +42,11 @@ spring.data.mongodb.port=27017
...
 #Flyway
 ####### Mongock Properties ###########
-mongock.change-logs-scan-package=com.knubisoft.application.mongo.config
\ No newline at end of file
+mongock.change-logs-scan-package=com.knubisoft.application.mongo.config
+###Emailing
+spring.mail.host=smtp.gmail.com
+spring.mail.port=587
+spring.mail.username=v.kolesnyk@knubisoft.com
+spring.mail.password=tufbvcsmvrhmkuwk
```
Comment (nikitakoliadin) [inline application/src/main/resources/application.properties]: Nice password :D

Comment (nikitakoliadin) [reply application/src/main/resources/application.properties]: Pls do research how to hide it.
For example you can encrypt it.

File: application/src/main/java/com/knubisoft/application/entity/EmailDetails.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import lombok.NoArgsConstructor;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class EmailDetails {
+    private String recipient;
+    private String msgBody;
+    private String subject;
+    private String attachment;
```
Comment (nikitashumsky) [inline application/src/main/java/com/knubisoft/application/entity/EmailDetails.java]: Do you want to implement List of attachments and restrict it by 3 files per message, something like that?

**Review by nikitakoliadin — CHANGES_REQUESTED** [review]
Waiting the tests

**Review by nikitakoliadin — APPROVED** [review]

File: application/src/main/java/com/knubisoft/application/service/EmailServiceImpl.java [hunk 5]
```diff
@@ -13,57 +13,62 @@
...
     @Autowired
     private JavaMailSender javaMailSender;
-
     @Value("${spring.mail.username}")
     private String sender;
+    private static final String MAIL_SENT_SUCCESS = "Mail Sent Successfully";
+    private static final String ERROR_SENDING_MAIL = "Error while sending mail";
+    private static final String ATTACHMENTS_CAP = "Attachments cap reached, only first three will be sent";
+    private static final String FILE_NOT_FOUND = "File by path: <%s> was not found in the system. "
+            + "Message will be sent will be sent without it.";
```
Comment (nikitashumsky) [inline application/src/main/java/com/knubisoft/application/service/EmailServiceImpl.java]: "will be sent" repeats 2 times

**Review by nikitashumsky — APPROVED** [review]

**Review by nikitakoliadin — APPROVED** [review]

**Review by connor999jobs — APPROVED** [review]

File: application/src/main/java/com/knubisoft/application/service/EmailServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+            prepareAttachments(emailDetails, mimeMessageHelper);
+        }
+    }
+
+    private void prepareAttachments(final EmailDetails emailDetails,
+                                    final MimeMessageHelper mimeMessageHelper) throws MessagingException {
+        if (emailDetails.getAttachments().size() > attachmentsCapSize) {
+            log.warn(ATTACHMENTS_CAP);
+        }
+        for (String attachment : emailDetails.getAttachments().subList(0, (attachmentsCapSize - 1))) {
```
Comment (nikitashumsky) [inline application/src/main/java/com/knubisoft/application/service/EmailServiceImpl.java]: sublist already doing (attachmentsCapSize - 1), cause he excludes the last index. Now we're getting 2\3 attachments when sending 3

**Pull request merged**